### PR TITLE
Add Integration Tests

### PR DESCRIPTION
**Pull request by hisaac**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

This Step hasn't been released yet, so no version change necessary.

### Context

Mimicking the PR for the `save-cache` Step (https://github.com/bitrise-steplib/bitrise-step-save-cache/pull/6), this change adds integration tests for the decompression code.

### Changes

- Moves decompression code to its own file
- Adds integration test file and test data archives
- Adds ability to pass extra arguments to the decompression command
    - I added this so that I could specify the directory to decompress the test archives to, in this case a temp directory

### CODE REVIEW

File: integration_tests/decompression_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
+//go:build integration
```
Comment (hisaac) [inline integration_tests/decompression_test.go:1]: GoLand told me I didn't need the second line, `// +build integration`, and it seems to work as expected without it.

Comment (ofalvai) [reply integration_tests/decompression_test.go:1]: Strange, I don't get a warning about the second line...but if it works it works!

File: integration_tests/decompression_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		},
+	}
+
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Given
+			logger := log.NewLogger()
+			envRepo := env.NewRepository()
+
+			tempDir, err := os.MkdirTemp("", "decompression_test")
```
Comment (ofalvai) [inline integration_tests/decompression_test.go:56]: I use to write the same test code until someone pointed out in a PR review that `t.TempDir()` does the same AND cleans itself up after the test

Comment (hisaac) [reply integration_tests/decompression_test.go]: Updated. Thanks for the tip, that's way nicer.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**